etc/gen_atlas_search/src/main.rs

```diff
@@ -0,0 +1,219 @@
+use convert_case::{Case, Casing};
+use proc_macro2::TokenStream;
+use quote::format_ident;
+use serde::Deserialize;
+use syn::parse_quote;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct Operator {
+    name: String,
+    #[expect(dead_code)]
+    link: String,
+    #[serde(rename = "type")]
+    #[expect(dead_code)]
+    type_: Vec<OperatorType>,
+    #[expect(dead_code)]
+    encode: EncodeType,
+    description: String,
+    arguments: Vec<Argument>,
+    tests: Vec<serde_yaml::Value>,
+}
+
+impl Operator {
+    fn clear_tests(mut self) -> Self {
+        self.tests.clear();
+        self
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum OperatorType {
+    SearchOperator,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum EncodeType {
+    Object,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+struct Argument {
+    name: String,
+    #[serde(default)]
+    optional: Option<bool>,
+    #[serde(rename = "type")]
+    type_: Vec<ArgumentType>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+enum ArgumentType {
+    String,
+    Object,
+    SearchScore,
+    SearchPath,
+    SearchOperator,
+    Array,
+    Int,
+}
+
+enum ArgumentRustType {
+    String,
+    Document,
+    StringOrArray,
+    TokenOrder,
+    MatchCriteria,
+    Operator,
+    I32,
+}
+
+static QUERY: &str = "query";
+static TOKEN_ORDER: &str = "tokenOrder";
+static MATCH_CRITERIA: &str = "matchCriteria";
+
+impl Argument {
+    fn rust_type(&self) -> ArgumentRustType {
+        if self.name == QUERY {
+            return ArgumentRustType::StringOrArray;
+        }
+        if self.name == TOKEN_ORDER {
+            return ArgumentRustType::TokenOrder;
+        }
+        if self.name == MATCH_CRITERIA {
+            return ArgumentRustType::MatchCriteria;
+        }
+        match self.type_.as_slice() {
+            [ArgumentType::String] => ArgumentRustType::String,
+            [ArgumentType::Object] => ArgumentRustType::Document,
+            [ArgumentType::SearchScore] => ArgumentRustType::Document,
+            [ArgumentType::SearchPath] => ArgumentRustType::StringOrArray,
+            [ArgumentType::SearchOperator, ArgumentType::Array] => ArgumentRustType::Operator,
+            [ArgumentType::Int] => ArgumentRustType::I32,
+            _ => panic!("Unexpected argument types: {:?}", self.type_),
```

Comment:
Since this is a standalone binary, I figure `panic!` / `unwrap` is actually reasonable error handling.